sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java

```diff
@@ -1478,56 +1456,25 @@ Mono<Response<QueueProperties>> createQueueWithResponse(String queueName, Create
      */
     Mono<Response<RuleProperties>> createRuleWithResponse(String topicName, String subscriptionName, String ruleName,
         CreateRuleOptions ruleOptions, Context context) {
-        if (topicName == null) {
-            return monoError(LOGGER, new NullPointerException("'topicName' cannot be null."));
-        } else if (topicName.isEmpty()) {
-            return monoError(LOGGER, new IllegalArgumentException("'topicName' cannot be empty."));
-        }
-
-        if (subscriptionName == null) {
-            return monoError(LOGGER, new NullPointerException("'subscriptionName' cannot be null."));
-        } else if (subscriptionName.isEmpty()) {
-            return monoError(LOGGER, new IllegalArgumentException("'subscriptionName' cannot be empty."));
-        }
-
-        if (ruleName == null) {
-            return monoError(LOGGER, new NullPointerException("'ruleName' cannot be null."));
-        } else if (ruleName.isEmpty()) {
-            return monoError(LOGGER, new IllegalArgumentException("'ruleName' cannot be empty."));
-        }
-
-        if (ruleOptions == null) {
-            return monoError(LOGGER, new NullPointerException("'rule' cannot be null."));
-        }
-
-        final RuleActionImpl action = ruleOptions.getAction() != null
-            ? EntityHelper.toImplementation(ruleOptions.getAction())
-            : null;
-        final RuleFilterImpl filter = ruleOptions.getFilter() != null
-            ? EntityHelper.toImplementation(ruleOptions.getFilter())
-            : null;
-        final RuleDescription rule = new RuleDescription()
-            .setAction(action)
-            .setFilter(filter)
-            .setName(ruleName);
-
-        final CreateRuleBodyContent content = new CreateRuleBodyContent()
-            .setType(CONTENT_TYPE)
-            .setRuleDescription(rule);
-        final CreateRuleBody createEntity = new CreateRuleBody().setContent(content);
-
-        final Context withTracing = context.addData(AZ_TRACING_NAMESPACE_KEY, AZ_TRACING_NAMESPACE_VALUE);
-
         try {
+            validateTopicName(topicName);
```

Comment:
My general rule about try/catch blocks is to limit the scope to where I think it may throw rather than wrapping everything in a try/catch and designing around that.  It think it may be a design preference at this point. I won't die on a hill if others think this is the way to go.

@anuchandy Thoughts?

Reply:
The typical pattern I've seen and trying to follow is to limit the scope of the try-catch. I wonder if we need these new entity helper methods to check null + empty for entity names, requiring uber try-catch?

Reply:
updated, to limit try/catch to the scope.